ledger/participant-integration-api/src/main/scala/platform/localstore/PersistentIdentityProviderConfigStore.scala

```diff
@@ -0,0 +1,172 @@
+
+import java.sql.Connection
+import scala.concurrent.{ExecutionContext, Future}
+
+class PersistentIdentityProviderConfigStore(
+    dbSupport: DbSupport,
+    metrics: Metrics,
+)(implicit executionContext: ExecutionContext)
+    extends IdentityProviderConfigStore {
+
+  private val backend = dbSupport.storageBackendFactory.createIdentityProviderConfigStorageBackend
+  private val dbDispatcher = dbSupport.dbDispatcher
+  private val logger = ContextualizedLogger.get(getClass)
+
+  override def createIdentityProviderConfig(identityProviderConfig: domain.IdentityProviderConfig)(
+      implicit loggingContext: LoggingContext
+  ): Future[Result[domain.IdentityProviderConfig]] = {
+    inTransaction(_.createIDPConfig) { implicit connection =>
+      val id = identityProviderConfig.identityProviderId
+      for {
+        _ <- idpConfigDoesNotExist(id)
+        _ <- idpConfigIssuerDoesNotExist(Some(identityProviderConfig.issuer))
+        _ = backend.createIdentityProviderConfig(identityProviderConfig)(connection)
+        domainConfig <- backend
+          .getIdentityProviderConfig(id)(connection)
+          .toRight(IdentityProviderConfigNotFound(id))
+      } yield domainConfig
+    }.map(tapSuccess { _ =>
+      logger.info(
+        s"Created new identity provider configuration: $identityProviderConfig"
+      )
+    })
+  }
+
+  override def getIdentityProviderConfig(id: IdentityProviderId.Id)(implicit
+      loggingContext: LoggingContext
+  ): Future[Result[domain.IdentityProviderConfig]] = {
+    inTransaction(_.getIDPConfig) { implicit connection =>
+      backend
+        .getIdentityProviderConfig(id)(connection)
+        .toRight(IdentityProviderConfigNotFound(id))
+    }
+  }
+
+  override def deleteIdentityProviderConfig(id: IdentityProviderId.Id)(implicit
+      loggingContext: LoggingContext
+  ): Future[Result[Unit]] = {
+    inTransaction(_.deleteIDPConfig) { implicit connection =>
+      if (!backend.deleteIdentityProviderConfig(id)(connection)) {
+        Left(IdentityProviderConfigNotFound(id))
+      } else {
+        Right(())
+      }
+    }.map(tapSuccess { _ =>
+      logger.info(
+        s"Deleted identity provider configuration with id $id"
+      )
+    })
+  }
+
+  override def listIdentityProviderConfigs()(implicit
+      loggingContext: LoggingContext
+  ): Future[Result[Seq[domain.IdentityProviderConfig]]] = {
+    inTransaction(_.listIDPConfigs) { implicit connection =>
+      Right(backend.listIdentityProviderConfigs()(connection))
+    }
+  }
+
+  override def updateIdentityProviderConfig(update: IdentityProviderConfigUpdate)(implicit
+      loggingContext: LoggingContext
+  ): Future[Result[domain.IdentityProviderConfig]] = {
+    inTransaction(_.updateIDPConfig) { implicit connection =>
+      val id = update.identityProviderId
+      for {
+        _ <- idpConfigExists(id)
+        _ <- idpConfigIssuerDoesNotExist(update.issuerUpdate)
+        _ = {
+          update.issuerUpdate.foreach(
```

Comment:
The very last call queries idp_config by id and it would give `IdentityProviderConfigNotFound` to the output, do you see a problem with this approach?

Reply:
A different transaction can delete and re-create the config while this one is running.
We probably cannot be totally bullet-proof (e.g. it all can happen after the last update and before the last query which fetches the record); but it seems prudent to check if the update queries updated anything at all